src/modules/drawing/DrawingModule.vue

```diff
@@ -1,50 +1,193 @@
 <template>
-    <div v-if="show" class="draw-overlay">
-        <DrawingToolbox
-            :drawing-modes="drawingModes"
-            :current-drawing-mode="currentDrawingMode"
-            @close="hideDrawingOverlay"
-            @setDrawingMode="changeDrawingMode"
-        />
+    <div>
+        <div v-if="show" class="draw-overlay">
+            <DrawingToolbox
+                :drawing-modes="drawingModes"
+                :current-drawing-mode="currentDrawingMode"
+                @close="hideDrawingOverlay"
+                @setDrawingMode="changeDrawingMode"
+            />
+        </div>
+        <div v-show="show">
+            <drawing-style-popup
+                ref="overlay"
+                :feature="selectedFeature"
+                @delete="deleteSelectedFeature"
+                @close="deactivateFeature"
+                @updateProperties="updateProperties"
+            />
+        </div>
     </div>
 </template>
 
 <script>
 import { mapState, mapActions } from 'vuex'
 import { drawingModes } from '@/modules/store/modules/drawing.store'
 import DrawingToolbox from '@/modules/drawing/components/DrawingToolbox'
+import DrawingManager from '@/modules/drawing/lib/DrawingManager'
+import { createEditingStyle } from '@/modules/drawing/lib/style'
+import DrawingStylePopup from './components/DrawingStylePopup.vue'
+import { Overlay } from 'ol'
+import { create, update } from '@/api/files.api'
+
+const overlay = new Overlay({})
 
 export default {
-    components: { DrawingToolbox },
+    components: { DrawingToolbox, DrawingStylePopup },
     inject: ['getMap'],
     computed: {
         ...mapState({
             show: (state) => state.ui.showDrawingOverlay,
             currentDrawingMode: (state) => state.drawing.mode,
+            geoJson: (state) => state.drawing.geoJson,
+            selectedFeature: function (state) {
+                const sfd = state.drawing.selectedFeatureData
+                const geoJson = state.drawing.geoJson
+                let f = null
+                if (sfd) {
+                    f = geoJson.features.find((f) => f.id === sfd.featureId)
+                }
+                overlay.setVisible(!!f)
+                if (!f) {
+                    return null
+                }
+                const xy = f.coordinate || [731667.39, 5862995.8]
+                overlay.setPosition(xy)
+                return f
+            },
```

Comment:
I would prefer that `mapState(...)` function is only used to extract information from the store, and does nothing else.

Please use a watcher if the visibility of the overlay needs to change depending on the state.